src/executor/insert.rs

```diff
@@ -85,21 +91,27 @@ where
 {
     type PrimaryKey<A> = <<A as ActiveModelTrait>::Entity as EntityTrait>::PrimaryKey;
     type ValueTypeOf<A> = <PrimaryKey<A> as PrimaryKeyTrait>::ValueType;
-    let last_insert_id = match db.get_database_backend() {
-        DbBackend::Postgres => {
+    let last_insert_id_opt = match db {
+        #[cfg(feature = "sqlx-postgres")]
+        DatabaseConnection::SqlxPostgresPoolConnection(conn) => {
```

Comment:
Oh. and here. the behavior has been changed since 85eadd38d222b07621ff5b5ca55b650295bbe682

Reply:
Let me merge from latest master and check